### DISCUSSION TITLE
Fastlane is suuuuuper slow to even start

### DISCUSSION THREAD

**Discussion by avioli** [fastlane/fastlane#21342] [General] [answered]
Every time I run fastlane it takes about 2 minutes for it even show any output:

<img width="812" alt="Screenshot 2023-06-15 at 10 55 07 am" src="https://github.com/fastlane/fastlane/assets/524259/ceaed95e-47e7-42c4-932f-84f0a7d42ed5">

In the above screenshot you can see the first output was from 10:52:46 and first relevant output after the troubleshoot confirmation was at 10:54:25, which is about 2 minutes of doing practically nothing!

My best guess is that it is loading a crap-load of `requires`, but I cannot debug it given that neither `--verbose`, nor `--troubleshoot` give any helpful output during that nap time.

Any help would be appreciated.

## Here's what I've done to install fastlane

I'm running on a Mac with Apple M1 Max CPU and 32 GB RAM if that is important.

Initially I've installed it via `brew install fastlane`, so I've got `fastlane: stable 2.213.0`. That installed `ruby@3.1` as well.

It was running slow, so I thought that it might be that it is "bottled" via brew, so I went ahead and uninstalled the brew version and re-installed it with `gem install fastlane`. My ruby is installed via `rbenv` and I am on `ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [arm64-darwin21]`. To confirm that I'm running the correct thing I run `which fastlane` I get `~/.rbenv/shims/fastlane`.

It was still running same slow - about 2 minutes to "boot".

I ran `gem cleanup` as well, but to no avail.

Tried running with `bundle exec fastlane ...` as seen in the screenshot with no change.

**Comment by avioli** [comment]
Got [ruby-prof-flamegraph](https://github.com/oozou/ruby-prof-flamegraph) profiling `fastlane --help` while running and got this output and flamegraph out of it:

[flamegraph.prof.zip](https://github.com/fastlane/fastlane/files/11753430/flamegraph.prof.zip)

This one is generated using [flamegraph.pl](https://github.com/brendangregg/FlameGraph) (installed via `brew install flamegraph`):

![flamegraph](https://github.com/fastlane/fastlane/assets/524259/37b1f603-9e9e-4c6d-afa8-0536288e3dc3)

It is interactive if you download the SVG and open it locally in your browser

All-in-all it shows there is some `Socket` connection iterating over `each_address` doing some `lazy_initialize`, then running `IO#each` over an `Array#+ (281684)`, which accounts for 97,365 samples - 83.78% out of all samples:

<img width="443" alt="Screenshot 2023-06-15 at 4 11 20 pm" src="https://github.com/fastlane/fastlane/assets/524259/fb7e4a9d-9e64-4c68-8dde-986c07a9b0d7">

I hope this helps somewhat.

**Reply by avioli** [reply]
Looking at the prof file shows that `Array#+ (281684)` took 97365.004539 ms, which is about 1 min 37 sec, which is exactly spot-on what I experience.

**Reply by avioli** [reply]
Running a grep on `start_looking_for_update` yields these:

```
lib/ruby/gems/3.1.0/gems/fastlane-2.213.0/fastlane/lib/fastlane/cli_tools_distributor.rb:        FastlaneCore::UpdateChecker.start_looking_for_update('fastlane')
lib/ruby/gems/3.1.0/gems/fastlane-2.213.0/fastlane/lib/fastlane/commands_generator.rb:        Fastlane::PluginUpdateManager.start_looking_for_updates
lib/ruby/gems/3.1.0/gems/fastlane-2.213.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:    def self.start_looking_for_updates
lib/ruby/gems/3.1.0/gems/fastlane-2.213.0/fastlane_core/lib/fastlane_core/update_checker/update_checker.rb:    def self.start_looking_for_update(gem_name)
```

Where the first hit has a comment above it:

```rb
# Needs to go after load_dot_env for variable FASTLANE_SKIP_UPDATE_CHECK
```

... so when I set that env variable `export FASTLANE_SKIP_UPDATE_CHECK=1` I get no delay and fastlane runs as expected!

**Comment by avioli** [comment, marked as answer]
My solution for the time being is to set `FASTLANE_SKIP_UPDATE_CHECK` env variable, since that turned out to be sidestepping the bottleneck in my case - a check for updates.

I've added this to my `.zshrc`:

```sh
export FASTLANE_SKIP_UPDATE_CHECK=1
```

**Reply by ladislas** [reply]
thanks! You saved my day :)
